models/issue_milestone.go

```diff
@@ -424,9 +424,12 @@ func GetMilestones(opts GetMilestonesOption) (MilestoneList, error) {
 }
 
 // SearchMilestones search milestones
-func SearchMilestones(repoCond builder.Cond, page int, isClosed bool, sortType string) (MilestoneList, error) {
+func SearchMilestones(repoCond builder.Cond, page int, isClosed bool, sortType string, keyword string) (MilestoneList, error) {
 	miles := make([]*Milestone, 0, setting.UI.IssuePagingNum)
 	sess := x.Where("is_closed = ?", isClosed)
+	if len(keyword) > 0 {
+		sess = sess.And(builder.Like{"name", keyword})
```

Comment:
This will be case-sensitive on many databases.

```suggestion
		sess = sess.And(builder.Like{"UPPER(name)", strings.ToUpper(keyword)})
```